### PR TITLE
Fix quality of SVG rendering in the online editor

### PR DESCRIPTION
**Pull request by tronical**
SVGs are rendered using HTML image elements, that are converted to textures. The size of the texture defaults to the SVGs viewbox, which may be small - despite it being possible to render the SVG at a higher resolution with great quality.

Similar to the native code path, this patch also uses the target image size and propagates it to the DOM HTML image element to instruct the browser to render the SVG at a higher resolution.

### CODE REVIEW

File: internal/core/graphics/image.rs [hunk 1]
```diff
@@ -435,6 +435,20 @@ impl ImageInner {
...
         }
     }
+
+    /// Returns true if the image is an SVG (either backed by resvg or HTML image wrapper).
+    pub fn is_svg(&self) -> bool {
+        match self {
+            Self::Svg(_) => true,
+            #[cfg(target_arch = "wasm32")]
+            Self::HTMLImage(html_image)
+                if html_image.dom_element.current_src().ends_with(".svg") =>
```
Comment (ogoffart) [inline internal/core/graphics/image.rs:445]: This line could be made into a function  `html_image.is_svg()`  so there is only one place with this .svg logic.  I wonder if there are no more reliable way to detect if an image is svg though  (mime type?)

**Review by ogoffart — APPROVED** [review]

File: internal/core/graphics/image.rs (hunk 1 above)
Comment (tronical) [reply internal/core/graphics/image.rs]: Good idea, done (the function part).

I wish, too, that there was a more reliable way. don't know of any though :(

File: internal/backends/winit/renderer/femtovg/images.rs [hunk 2]
```diff
@@ -101,6 +101,11 @@ impl Texture {
                     // pre-multipled alpha. It's possible that this is not generally applicable, but it
                     // is the case for SVGs.
                     let image_flags = if html_image.dom_element.current_src().ends_with(".svg") {
```
Comment (ogoffart) [inline internal/backends/winit/renderer/femtovg/images.rs:103]:
```suggestion
                    let image_flags = if html_image.is_svg() {
```

(that was the whole point of moving it to a function)

Comment (tronical) [reply internal/backends/winit/renderer/femtovg/images.rs]: 🤦‍♂️ thanks!

**Pull request merged**